image_view/src/disparity_view_node.cpp

```diff
@@ -31,127 +31,92 @@
-static void destroyNode(GtkWidget *widget, gpointer data)
-{
-  exit(0);
-}
+#include <gtk/gtk.h>
 
-static void destroyNodelet(GtkWidget *widget, gpointer data)
-{
-  // We can't actually unload the nodelet from here, but we can at least
-  // unsubscribe from the image topic.
-  reinterpret_cast<ros::Subscriber*>(data)->shutdown();
-}
 #endif
 
-
-namespace image_view {
-
-class DisparityNodelet : public nodelet::Nodelet
+namespace image_view
 {
-  // colormap for disparities, RGB order
-  static unsigned char colormap[];
-
-  std::string window_name_;
-  ros::Subscriber sub_;
-  cv::Mat_<cv::Vec3b> disparity_color_;
-  bool initialized;
-  
-  virtual void onInit();
-  
-  void imageCb(const stereo_msgs::DisparityImageConstPtr& msg);
-
-public:
-  ~DisparityNodelet();
-};
 
-DisparityNodelet::~DisparityNodelet()
+DisparityViewNode::DisparityViewNode(const rclcpp::NodeOptions & options)
+  : rclcpp::Node("disparity_view_node", options)
 {
-  cv::destroyWindow(window_name_);
-}
+  std::string topic = rclcpp::expand_topic_or_service_name("image", this->get_name(), this->get_namespace());
 
-void DisparityNodelet::onInit()
-{
-  initialized = false;
-  ros::NodeHandle nh = getNodeHandle();
-  ros::NodeHandle local_nh = getPrivateNodeHandle();
-  const std::vector<std::string>& argv = getMyArgv();
+  if (topic == "image") {
+    RCLCPP_WARN(
+      this->get_logger(), "Topic 'image' has not been remapped! Typical command-line usage:\n"
+      "\t$ rosrun image_view disparity_view image:=<disparity image topic>");
+  }
 
-  // Internal option, should be used only by image_view nodes
-  bool shutdown_on_close = std::find(argv.begin(), argv.end(),
-                                     "--shutdown-on-close") != argv.end();
+  initialized = false;
 
   // Default window name is the resolved topic name
-  std::string topic = nh.resolveName("image");
-  local_nh.param("window_name", window_name_, topic);
+  window_name_ = this->declare_parameter("window_name", topic);
+  // bool autosize = this->declare_parameter("autosize", false);
 
-  bool autosize;
-  local_nh.param("autosize", autosize, false);
+  // cv::namedWindow(window_name_, autosize ? cv::WND_PROP_AUTOSIZE : 0);
 
-  //cv::namedWindow(window_name_, autosize ? cv::WND_PROP_AUTOSIZE : 0);
-#if CV_MAJOR_VERSION ==2
-#ifdef HAVE_GTK
-  // Register appropriate handler for when user closes the display window
-  GtkWidget *widget = GTK_WIDGET( cvGetWindowHandle(window_name_.c_str()) );
-  if (shutdown_on_close)
-    g_signal_connect(widget, "destroy", G_CALLBACK(destroyNode), NULL);
-  else
-    g_signal_connect(widget, "destroy", G_CALLBACK(destroyNodelet), &sub_);
-#endif
-  // Start the OpenCV window thread so we don't have to waitKey() somewhere
-  startWindowThread();
-#endif
+  sub_ = this->create_subscription<stereo_msgs::msg::DisparityImage>(
+    topic, rclcpp::QoS(10), std::bind(&DisparityViewNode::imageCb, this, std::placeholders::_1));
```

Comment:
Curious - why 10?

Reply:
10 is the default queue size before providing an `rclcpp::QoS()` was required. It just explicitly uses the default.